include/seqan3/range/decorator/gap_decorator.hpp

```diff
@@ -507,7 +505,7 @@ class gap_decorator
      * ### Exceptions
      * No-throw guarantee.
      */
-    constexpr size_type gap_length(set_iterator_type it) const noexcept
+    size_type gap_length(set_iterator_type it) const noexcept
```

Comment:
and this is now guaranteed nothrow?

Reply:
I thought so... the code is
```cpp
return (it == anchors.begin()) ? it->second : it->second - (*std::prev(it)).second;
```
`anchors.begin()` is documented as `noexcept` and `std::prev` is constexpr so it is also noecept because constexpr function cannot throw. 
The iterator of the set is comparing and dereferencing a pair of two integers, so since working with PODs this should also be safe?

Reply:
constexpr modifier does not imply noexcept when the execution is done at runtime

https://godbolt.org/z/o1FBYD

Reply:
Oh, very interesting. Then I will remove the noexcept here too!